editor/src/app/sites/site.component.ts

```diff
@@ -83,7 +85,17 @@ export class SiteComponent implements OnInit {
 
   updateField(field: string, value: string) {
     if (field === 'name') {
-      this.store.dispatch(new RenameSiteAction(this.site, value));
+      this.store.dispatch(new RenameSiteAction(this.site, value)).subscribe({
```

Comment:
The idea is good. But we should clean up the stream, instead of subscription please use `switchMap`. Also don't use `this.site.name`, use the new one from the state - becaue it may have changed from the sync.
```ts
this.store.dispatch(new RenameSiteAction(this.site, value)).pipe(
    map(state => {
        return state.sites.find(site => site.order === this.site.order);
    }),
    switchMap(renamedSite => {
        this.route.queryParams.pipe(
            take(1),
            filter(params => params.site && params.site === renamedSite.name)
        )
    })
).subscribe((renamedSite) => {
    this.router.navigate([], { queryParams: { site: renamedSite.name }, queryParamsHandling: 'merge' });
});
```